### DISCUSSION TITLE
Any plans to take this to GUI? Also, splitting package up?

### DISCUSSION THREAD

**Discussion by liamzee** [jtdaugherty/brick#481] [General]
I'm familiar with another UI library, that seems to be, well, a bit undermaintained. One suggestion (that got shot down) was splitting the library up so that the execution space and the widgets were split into separate packages, to encourage custom widget creation and separate package creation.

Another element is, well, brick is the standard Haskell TUI library. Is there any interest in building a GUI edition for it, sharing the same widget system?

**Comment by jtdaugherty** [comment]
> Is there any interest in building a GUI edition for it, sharing the same widget system?

No, Brick has been and will always be for terminals. That was, after all, the reason I built it, and I would recommend that if people want to build applications with GUIs then there are lots of better libraries for that!

On the topic of the "widget" notion in Brick, its "widgets" are not quite the same kind of thing as one finds in typical GUI libraries. In typical GUI libraries, "widget" is the name given to something that has an on-screen representation *and* its state *and* its event handling machinery. In Brick, all three of those things are separate and the "widget" is *only* its on-screen representation.

**Comment by liamzee** [comment]
Thank you so much for the response.

As far as GUIs go, gi-gtk-declarative seems dead, monomer is a bit slow in development, so I guess I should give wxwidgets a try (new maintainer, I think). The GUI point is more that Brick is a well-established and goto library in Haskell, and it's declarative.

Thank you so much for the library; it's an absolutely great cornerstone.

By the way, if someone made a GUI library and claimed it was inspired by Brick, would you have issues with that?

**Reply by jtdaugherty** [reply]
> Thank you so much for the library; it's an absolutely great cornerstone.

I'm glad you're finding it useful!

> By the way, if someone made a GUI library and claimed it was inspired by Brick, would you have issues with that?

No, that would frankly be pretty cool as far as I'm concerned! Brick itself was inspired by [gloss](https://hackage.haskell.org/package/gloss), at least, Gloss circa 2015 when I started building Brick. :)
